File: house/controllers/build_controller.py
from house.controllers.basic_controller import *
import json

class BuildController(BasicController):
    """docstring for LoadController."""
    def __init__(self, args):
        super().__init__(args)

        # Loading configuration file
        configpath = 'src/{0}/.houseconfig'.format(self.args[0])
        with open(configpath, 'r') as fp:
            content = fp.read()
        self.params = json.loads(content)['build']

    def draw(self):
        commands = []
        dirs = self.args[0].split('/')
        build_command = self.params['command']

        # Checking if build is local
        if 'local' in self.params:
            if self.params['local']:
                commands.append('cd src')
                for direc in dirs:
                    commands.append('cd ' + direc)
                commands.append(build_command)
                levels = 1 + len(dirs)
                for _ in range(levels):
                    commands.append('cd ..')
            else:
                commands.append(build_command)
        else:
            commands.append(build_command)

        # Executing commands
        self.execute(commands)

File: house/controllers/build_controller.lua
local basic_controller = require "house/controllers/basic_controller"
local JSON = require "dkjson"
local build_controller = { }

build_controller.construct = function(args)
  local self = basic_controller.new(args)

  -- Loading configuration file
  local configpath = '.'
  if self.repo ~= nil then
    configpath = 'src/' .. self.repo
  end
  configpath = configpath .. '/.houseconfig'
  local config = util.readAll(configpath)
  self.params = JSON.decode(config, 1, nil).build

  return self
end

build_controller.new = function(args)
  local self = build_controller.construct(args)

  self.draw = function()
    local commands = { }

    if (self.params['local'] == true) and (self.repo ~= nil) then
      local dirs = util.mysplit(self.repo, '/')
      table.insert(commands, 'cd src')
      for _, d in ipairs(dirs) do
        table.insert(commands, 'cd ' .. d)
      end
      table.insert(commands, self.params.command)
      for _ = 1, 1 + #dirs do
        table.insert(commands, 'cd ..')
      end
    else
      table.insert(commands, self.params.command)
    end

    self.execute(commands)
  end

  return self
end

return build_controller
